iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -72,7 +72,11 @@ class StaticString {
       public:
         ~UncheckedConstAccessor() = default;
         UncheckedConstAccessor(UncheckedConstAccessor const&) = delete;
+#if __cplusplus >= 201703L
         UncheckedConstAccessor(UncheckedConstAccessor&&) = delete;
+#else
+        UncheckedConstAccessor(UncheckedConstAccessor&&) = default;
+#endif
```

Comment:
@ComicSansMS I had to implement the move c'tor since RVO is not mandatory in C++14. I think it should not cause any issues. If we want to prevent that the move c'tor is used, I could add an `assert(false)` in the body of the implementation.

I'm not sure about the compiler switch, though. Although the string is header only and this is a temporary object, we might end up in having the C++ bindings compiled with C++14 and user code using C++17 or newer. This might end up in some hard to track issues. Shall we introduce a `deployment.hpp.in` and "hardcode" the value that is used to build this lib?

Reply:
Okay, I will then remove the compiler switch and add a comment to change the `default` to `delete` when we move to C++17 as min version.